src/D2L.Bmx/UpdateChecker.cs

```diff
@@ -0,0 +1,119 @@
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace D2L.Bmx;
+internal class UpdateChecker {
+
+	public static async Task CheckForUpdatesAsync( BmxConfig config ) {
+		try {
+			string? savedLatestVersion = GetSavedLatestVersion();
+			var localVersion = Assembly.GetExecutingAssembly().GetName().Version;
+			var latestVersion = new Version( savedLatestVersion ?? "0.0.0" );
+			if( ShouldFetchLatestVersion( savedLatestVersion ) ) {
+				latestVersion = new Version( await GetLatestReleaseVersionAsync() );
+			}
+
+			string updateLocation = ( string.Equals( config.Org, "d2l", StringComparison.OrdinalIgnoreCase ) )
+				? "https://bmx.d2l.dev"
+				: "https://github.com/Brightspace/bmx/releases/latest";
+
+			if( latestVersion > localVersion ) {
+				DisplayUpdateMessage( $"A new BMX release is available: v{latestVersion} (current: v{localVersion})\n" +
+					$"Upgrade now at {updateLocation}" );
+			}
+		} catch( Exception ex ) {
```

Comment:
```suggestion
		} catch( Exception ) {
```
Convention for making it clear you are catching an exception but don't care about the exception itself.

Reply:
Ooh thanks